master-branch-checker/pipe.sh

```diff
@@ -1,20 +1,22 @@
 #!/bin/sh
 set -e
 
-DIFF_FROM_MASTER=$(git rev-list --left-right --count HEAD...origin/master|awk '{print $2}')
+BRANCH_NAME=$(git branch | cut -c 3- | grep -E '^master$|^main$')
```

Comment:
Can we use git branch -r | grep -Po 'HEAD -> .*/\K.*$' instead ?

Reply:
Not sure if it's just me but I'm not getting any output from that. None of the branches are prefixed with 'HEAD ->' for me when running `git branch -r`

Reply:
Which code base are you on ? Suggesting that as it choses the remotes default branch so it can be called v2 for example and still be picked up as the main branch

Reply:
Partner Data Service. And yeah I started looking for a way to not assume the 'master' branch name I came across using `git remote show origin | grep -Po 'HEAD branch: \K.*$'` but it takes a few seconds due to having to contact the origin